doc/guides/porting/arch.rst

```diff
@@ -460,8 +460,97 @@ be derived from the linker scripts of other architectures. Some sections might
 be specific to the new architecture, for example the SCB section on ARM and the
 IDT section on x86.
 
-Hardware Stack Protection
-=========================
+Memory Management
+*****************
+
+If the target platform enables paging and requires drivers to memory-map
+their I/O regions, :option:`CONFIG_MMU` needs to be enabled and the
+:cpp:func:`arch_mem_map()` API implemented.
+
+Stack Objects
+*************
+
+The presence of memory protection hardware affects how stack objects are
+created. All architecture ports must specify the required alignment of the
+stack pointer, which is some combination of CPU and ABI requirements. This
+is defined in architecture headers with :c:macro:`ARCH_STACK_PTR_ALIGN` and
+is typically something small like 4, 8, or 16 bytes.
+
+Two types of thread stacks exist:
+
+- "kernel" stacks defined with :c:macro:`K_KERNEL_STACK_DEFINE()` and related
+  APIs, which can host kernel threads running in supervisor mode or
+  used as the stack for interrupt/exception handling. These have significantly
+  relaxed alignment requirements and use less reserved data.
```

Comment:
Well, actually it is not only that: the main point is that the script that generates a privilege stack for each defined thread stack _skips_ doing it for kernel stacks. Maybe worth mentioning.

Reply:
now reads: "These have significantly relaxed alignment requirements and use less reserved data. No memory is reserved for prvilege elevation stacks."